extensions/markdown-language-features/src/features/documentLinkProvider.ts

```diff
@@ -85,6 +85,20 @@ export default class LinkProvider implements vscode.DocumentLinkProvider {
 			} catch (e) {
 				// noop
 			}
+			if (match[5]) {
+				const imagePre = match[3];
+				const imageLink = match[5];
+				const imageOffset = (match.index || 0) + imagePre.length + 1;
```

Comment:
I agree that you can refactor out this logic. I'd create a function that either returns a `DocumentLink` or `undefined` (if an exception was thrown)